temoa/_internal/table_writer.py

```
@@ -81,13 +76,64 @@ def __init__(self, config: TemoaConfig, epsilon: float = 1e-5) -> None:
+        # Intersection: keys present in data AND present in database table
+        target_columns = list(data_keys.intersection(valid_columns))
+        target_columns.sort()  # Sort to ensure consistent order
+
+        if not target_columns:
+            logger.warning('No matching columns found for table %s. Skipping insert.', table_name)
+            return
+
+        # Prepare SQL statement
+        cols_str = ', '.join(target_columns)
+        placeholders = ', '.join(['?'] * len(target_columns))
+        query = f'INSERT INTO {table_name} ({cols_str}) VALUES ({placeholders})'
```

Ruff (0.14.8)

Possible SQL injection vector through string-based query construction

(S608)